admin/src/services/presetWidget.js

```diff
@@ -29,8 +29,14 @@ export class PresetWidgetService extends AbstractWidgetService {
     createAndShowWithTtl(presetSettings, ttlMs) {
         return this.apiPost("/create_and_show_with_ttl?ttl=" + ttlMs, presetSettings).catch((e) => this.errorHandler("Failed to add preset")(e));
     }
+
+    getPreview(id) {
+        return this.apiPost("/" + id + "/preview", undefined, "GET")
```

Comment:
Potentialy vulnerable to CSRF. We should check if we are not giving access to 'id' field to the user in the URL.
It's only ok to concatenate URLs if we are sure that the data is sane and sanitized and does not come from the user in any way (eg. does not appear in the webpage url/url query params/anywhere that a user can edit it.)
One link that is sent to the live crew could show something nasty.